lua/lspconfig/server_configurations/leanls.lua

```diff
@@ -20,6 +24,19 @@ return {
         or util.find_git_ancestor(fname)
     end,
     on_new_config = function(config, root_dir)
+      local lake_version = ''
+      local lake_job = vim.fn.jobstart({ 'lake', '--version' }, {
+        on_stdout = function(_, d, _)
+          lake_version = table.concat(d, '\n')
+        end,
+        stdout_buffered = true,
+      })
+      if lake_job > 0 and vim.fn.jobwait({ lake_job })[1] == 0 then
```

Comment:
If `lake --version` fails to start, we should also fall back to the legacy command (for Lean 4 versions before lake existed).

Reply:
You still need to use versions that old?

Reply:
Yes?  We're already checking for the status code, so we might as well fall back correctly.

Reply:
OK I suppose -- it's less clear to me that that's more "correct" unless it's really prevalent to deal with those versions -- the deterministic behavior seemed more valuable. E.g. if for some transient reason (out of PIDs momentarily, say) `lean --version` failed, we'll fallback "incorrectly", hence why I asked whether the likelihood someone is on something like `4.0.0-m2` is bigger. But I can change.

Reply:
I think it's very unlikely that `lake --version` fails but then `lean --server` works afterwards.